app/src/main/java/org/thoughtcrime/securesms/ui/AlertDialog.kt

```diff
@@ -32,53 +42,59 @@ class DialogButtonModel(
     val onClick: () -> Unit = {},
 )
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AlertDialog(
     onDismissRequest: () -> Unit,
     title: String? = null,
     text: String? = null,
-    content: @Composable () -> Unit = {},
-    buttons: List<DialogButtonModel>? = null
+    buttons: List<DialogButtonModel>? = null,
+    showCloseButton: Boolean = false,
+    content: @Composable () -> Unit = {}
 ) {
-    androidx.compose.material.AlertDialog(
-        onDismissRequest,
-        shape = MaterialTheme.shapes.small,
-        backgroundColor = LocalColors.current.backgroundSecondary,
-        buttons = {
-            Box {
-                IconButton(
-                    onClick = onDismissRequest,
-                    modifier = Modifier.align(Alignment.TopEnd)
-                ) {
-                    Icon(
-                        painter = painterResource(id = R.drawable.ic_dialog_x),
-                        tint = LocalColors.current.text,
-                        contentDescription = "back"
-                    )
+    BasicAlertDialog(
+        onDismissRequest = onDismissRequest,
+        content = {
+            Box(
+                modifier = Modifier.background(color = LocalColors.current.backgroundSecondary,
+                    shape = MaterialTheme.shapes.small)
+            ) {
+                // only show the 'x' button is required
+                if(showCloseButton) {
+                    IconButton(
+                        onClick = onDismissRequest,
+                        modifier = Modifier.align(Alignment.TopEnd)
+                    ) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.ic_dialog_x),
+                            tint = LocalColors.current.text,
+                            contentDescription = "back"
+                        )
+                    }
                 }
```

Comment:
Do you like this format? IDK if it is easy to read for everyone.

```suggestion
                if(showCloseButton) IconButton(
                    onClick = onDismissRequest,
                    modifier = Modifier.align(Alignment.TopEnd)
                ) {
                    Icon(
                        painter = painterResource(id = R.drawable.ic_dialog_x),
                        tint = LocalColors.current.text,
                        contentDescription = "back"
                    )
                }
            }
```

Reply:
I personally find it more readable with the new line. It feels clearer to me as opposed to lining up the first composable declaration on the same line as the `if`.
But happy to go with the team's preference on this.

Reply:
I think agree with you on this one, sometimes it looks better.